cpp/basix/lattice.cpp

```diff
@@ -18,58 +18,54 @@ using namespace basix;
 namespace
 {
 //-----------------------------------------------------------------------------
-xt::xtensor<double, 1> create_interval_equispaced(int n, bool exterior)
+xt::xtensor<double, 2> create_interval_equispaced(int n, bool exterior)
 {
-  if (exterior)
-    return xt::linspace<double>(0.0, 1.0, n + 1);
-  else
-  {
-    const double h = 1.0 / static_cast<double>(n);
-    return xt::linspace<double>(h, 1.0 - h, n - 1);
-  }
+  const double h = exterior ? 0 : 1.0 / static_cast<double>(n);
+  const std::size_t num_pts = exterior ? n + 1 : n - 1;
+  xt::xtensor<double, 1> pts = xt::linspace<double>(h, 1.0 - h, num_pts);
+  return xt::reshape_view(pts, {num_pts, (std::size_t)1});
 }
 //-----------------------------------------------------------------------------
-xt::xtensor<double, 1> create_interval_gll(int n, bool exterior)
+xt::xtensor<double, 2> create_interval_gll(int n, bool exterior)
 {
   if (n == 0)
-    return {0.5};
+    return {{0.5}};
 
-  auto _pts = quadrature::get_gll_points(n + 1);
+  const xt::xtensor<double, 2> _pts = quadrature::get_gll_points(n + 1);
 
   const std::size_t b = exterior ? 0 : 1;
-  std::array<std::size_t, 1> s = {static_cast<std::size_t>(n + 1 - 2 * b)};
-  xt::xtensor<double, 1> x(s);
-
+  std::array<std::size_t, 2> s
+      = {static_cast<std::size_t>(n + 1 - 2 * b), (std::size_t)1};
+  xt::xtensor<double, 2> x(s);
   if (exterior)
   {
-    x(0) = _pts[0];
-    x(n) = _pts[1];
+    x(0, 0) = _pts(0, 0);
+    x(n, 0) = _pts(1, 0);
   }
-
   for (std::size_t j = 2; j < static_cast<std::size_t>(n + 1); ++j)
-    x(j - 1 - b) = _pts[j];
-
+    x(j - 1 - b, 0) = _pts(j, 0);
   return x;
 }
 //-----------------------------------------------------------------------------
-xt::xtensor<double, 1> create_interval_chebyshev(int n, bool exterior)
+xt::xtensor<double, 2> create_interval_chebyshev(int n, bool exterior)
 {
   if (exterior)
   {
     throw std::runtime_error(
         "Chebyshev points including endpoints are not supported.");
   }
 
-  std::array<std::size_t, 1> s = {static_cast<std::size_t>(n - 1)};
-  xt::xtensor<double, 1> x(s);
+  std::array<std::size_t, 2> s
+      = {static_cast<std::size_t>(n - 1), (std::size_t)1};
```

Comment:
Would this still work if it was `      = {static_cast<std::size_t>(n - 1), 1};`?